packages/server/modules/comments/services/index.js

```diff
@@ -1,12 +1,55 @@
 'use strict'
 const crs = require('crypto-random-string')
 const knex = require('@/db/knex')
+const { Forbidden } = require('@/modules/shared/errors')
 const sanitizeHtml = require('sanitize-html')
 
 const Comments = () => knex('comments')
 const CommentLinks = () => knex('comment_links')
 const CommentViews = () => knex('comment_views')
 
+const resourceCheck = async (res, streamId) => {
+  // The switch of doom: if something throws, we're out
+  switch (res.resourceType) {
+    case 'stream':
+      // Stream validity is already checked, so we can just go ahead.
+      break
+    case 'commit': {
+      const linkage = await knex('stream_commits')
+        .select()
+        .where({ commitId: res.resourceId, streamId })
+        .first()
+      if (!linkage) throw new Error('Commit not found')
```

Comment:
Ideally we'd want to use custom error subclasses, so that we can ensure our `catch` clauses only catch what we intend to catch and that we don't have to match errors by their messages which is easy to break